aptos-db: `revert_commit` reordering and improvements

mzabaluev opened the pull request:
### Description

Change the order of database updates when reverting to a past version in the `revert_commit` implementation.
Remove KV data from some important schemas, document debt on removing data from others.

### Test Plan

`cargo test -p aptos-db`

There are very minimal tests covering `revert_commit`, will try to verify that more values were reverted.

mzabaluev committed:
fix(aptos-db): set in-memory ledger info on revert

mzabaluev committed:
refactor(aptos-db): use truncation helpers

Reuse the already existing truncation helpers to revert the
storage to an earlier version when reverting a commit.

mzabaluev committed:
fix(aptos-db): use the state store lock on revert

mzabaluev committed:
fix(aptos-db): use get_synced_version on revert

0xmovses commented on storage/aptosdb/src/db/include/aptosdb_writer.rs:
```diff
@@ -217,55 +216,38 @@ impl DbWriter for AptosDB {
-            target_version + 1,
-            latest_version + 1,
-            &batch,
+        truncate_ledger_db(self.ledger_db.clone(), target_version)?;
```
So truncate is the preferred way here .. yes, seems much safer.

0xmovses reviewed (commented):
Have you checked if the unit tests pass?

https://github.com/movementlabsxyz/aptos-core/blob/7df2b52fad033b853dc09981ed4ac12f1c18c035/storage/aptosdb/src/db/aptosdb_test.rs#L214

And: https://github.com/movementlabsxyz/aptos-core/blob/7df2b52fad033b853dc09981ed4ac12f1c18c035/storage/aptosdb/src/db/aptosdb_test.rs#L269

mzabaluev commented:
> Have you checked if the unit tests pass?

These tests do; I put some more checks there to test we're really removing things, but did not have much time to put more.
Please suggest some ideas on checking things we're interested to be cleaned up. I'll add checks for at least the block infos, as I have written code to revert them (it went in already with #30).

mzabaluev commented:
> I'll add checks for at least the block infos, as I have written code to revert them

Storing block infos is conditionally enabled when there is no indexer, so we don't actually store them with the store configuration that we use in movement, I believe. I'm looking at this for integration, so we may need to enable it.

mzabaluev committed:
fix(aptos-db): reorder reset on revert

l-monninger commented:
@mzabaluev this is still a draft actually, right?

mzabaluev commented:
It's fair to convert this into a draft as I'm debugging the block info effects.

mzabaluev committed:
test(aptos-db): improve revert tests

Check the tx events have been removed. This should
include new block events, but we can't directly test
that because the test block data generator used in aptos-db
does not generate BlockMetadata transactions.

Make variable naming more consistent in the tests.

l-monninger reviewed (approved):
Passing tests. LGTM otherwise too.
